tripzone/src/containers/FlightSearch/FlightSearch.js

```diff
@@ -0,0 +1,151 @@
+  const [sourceCode, setSourceCode] = useState("");
+  const [destinationCode, setDestinationCode] = useState("");
+  const [flightData, setFlightData] = useState([]);
+  const [showBookCard, setShowBookCard] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const [cities, setCities] = useState([]);
+  const [selectedFlight, setSelectedFlight] = useState(null);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  const {FLIGHT_SEARCH} = AppConstants.COMMONS;
+ 
+  // Function used to handle when a city is Searched
+
+  const handleSearch = async (e) => {
+    e.preventDefault();
+    setIsLoading(true);
+    setError(null);
+    try {
+      const data = await FlightService(sourceCode, destinationCode);
+      setFlightData(data);
+      setIsSubmitted(true);
+      setShowBookCard(false);
+    } catch (error) {
+      setError(error.message);
+    }
+    setIsLoading(false);
+  };
+
+
+  // Used to fetch cities data
+
+  useEffect(() => {
+    async function fetchCities() {
+      try {
+        const data = await getCities();
+        setCities(data);
+        setIsLoading(false);
+      } catch (e) {
+        setError(e);
+      }
+    }
+    fetchCities();
+  }, []);
+
+  const handleDestinationChange = (e) => {
+    setDestinationCode(e.target.value);
+    setDestination(e.target.value);
+  }
+  return (
+    <div>
+      <div className="form-container">
+
+        {/* FLIGHT SEARCH CONTAINER STARTS  */}
+
+        <form className="flight-form" onSubmit={handleSearch}>
+          <h3 className="flight-form-header">{FLIGHT_SEARCH.FORM_HEADER}</h3>
+
+          <p className="label">{FLIGHT_SEARCH.SOURCE}</p>
+          <Dropdown
+            id="sourceCode"
+            name="sourceCode"
+            value={sourceCode}
+            onChange={(event) => setSourceCode(event.target.value)}
+            className="form-input"
+            data={cities}
+            required
+          ></Dropdown>
+          <br />
+          <p className="label">{FLIGHT_SEARCH.DESTINATION}</p>
+          <Dropdown
+            id="destinationCode"
+            name="destinationCode"
+            value={destinationCode}
+            onChange={handleDestinationChange}
+            className="form-input"
+            data={cities}
+            required
+          ></Dropdown>
+          <br />
+          <Button
+            size="large"
+            type="submit"
+            value={isLoading ? "Searching..." : "SEARCH"}
+          ></Button>
+        </form>
+        {/* FLIGHT SEARCH CONTAINER ENDS  */}
+        
+        {/* AVAILABLE FLIGHT CONTAINER STARTS  */}
+
+        {error && <div>Error: {error}</div>}
+        {isSubmitted && flightData.length > 0 ? (
+          <div className="available-flights-container">
+            <h3 className="available-flights-header">{FLIGHT_SEARCH.AVAILABLE_FLIGHTS_HEADER}</h3>
+            <div>
+              {flightData.map((flight) => (
```

Comment:
here flightData.map is working , in other place, could you tell me why [spotData].map is needed

Reply:
Format of flightData is a JSON array, whereas spotData is a JSON object